### PR TITLE
Enable nullability in DataGridViewRowHeaderCell

### PR DESCRIPTION
**Pull request by gpetrou**
## Proposed changes

- Enable nullability in DataGridViewRowHeaderCell.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/10120)

### CODE REVIEW

File: src/System.Windows.Forms/src/System/Windows/Forms/DataGridViewRowHeaderCell.cs [hunk 1]
```diff
@@ -327,71 +267,65 @@ protected override Rectangle GetErrorIconBounds(Graphics graphics, DataGridViewC
...
-            return DataGridView.ContextMenuStrip;
-        }
-        else
-        {
-            return null;
-        }
+        ContextMenuStrip? contextMenuStrip = GetContextMenuStrip(rowIndex);
+        return contextMenuStrip is not null
+            ? contextMenuStrip
+            : DataGridView?.ContextMenuStrip;
```
Comment (halgab) [inline src/System.Windows.Forms/src/System/Windows/Forms/DataGridViewRowHeaderCell.cs:319]:
```suggestion
        return contextMenuStrip ?? DataGridView?.ContextMenuStrip;
```

File: src/System.Windows.Forms/src/System/Windows/Forms/DataGridViewRowHeaderCell.cs [hunk 2]
```diff
@@ -619,24 +558,25 @@ protected override Size GetPreferredSize(Graphics graphics, DataGridViewCellStyl
         }
 
         // Intentionally not using GetFormattedValue because header cells don't typically perform formatting.
-        object val = GetValue(rowIndex);
-        if (!(val is string))
+        object? val = GetValue(rowIndex);
+        if (val is not string)
         {
             val = null;
         }
```
Comment (halgab) [inline src/System.Windows.Forms/src/System/Windows/Forms/DataGridViewRowHeaderCell.cs:565]:
```suggestion
        string? val = GetValue(rowIndex) as string;
```

**Comment by gpetrou** [comment]
@lonitra any thoughts on this one?

**Review by lonitra — APPROVED** [review]
LGTM

**Pull request merged**